src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -170,25 +177,93 @@ private void loadData(Executor executor) throws IllegalStateException {
     }
   }
 
+  /**
+   * load ZoneId.SHORT_IDS and map to time zone index in transition table.
+   */
+  private void loadTimeZoneShortIDs(Map<String, Integer> zoneIdToTable) {
+    HostColumnVector.DataType type = new HostColumnVector.StructType(false,
+    new HostColumnVector.BasicType(false, DType.STRING),
+    new HostColumnVector.BasicType(false, DType.INT32));
+    ArrayList<HostColumnVector.StructData> data = new ArrayList<>();
+    // copy short IDs
+    List<String> idList = new ArrayList<>(ZoneId.SHORT_IDS.keySet());
+    // sort short IDs
+    Collections.sort(idList);
+    for (String id : idList) {
+      String mapTo = ZoneId.SHORT_IDS.get(id);
+      if (mapTo.startsWith("+") || mapTo.startsWith("-")) {
+        // skip: EST: -05:00; HST: -10:00; MST: -07:00
+        // kernel will handle EST, HST, MST
```

Comment:
```java  
static {
        Map<String, String> map = new HashMap<>(64);
        map.put("ACT", "Australia/Darwin");
        map.put("AET", "Australia/Sydney");
        map.put("AGT", "America/Argentina/Buenos_Aires");
        map.put("ART", "Africa/Cairo");
        map.put("AST", "America/Anchorage");
        map.put("BET", "America/Sao_Paulo");
        map.put("BST", "Asia/Dhaka");
        map.put("CAT", "Africa/Harare");
        map.put("CNT", "America/St_Johns");
        map.put("CST", "America/Chicago");
        map.put("CTT", "Asia/Shanghai");
        map.put("EAT", "Africa/Addis_Ababa");
        map.put("ECT", "Europe/Paris");
        map.put("IET", "America/Indiana/Indianapolis");
        map.put("IST", "Asia/Kolkata");
        map.put("JST", "Asia/Tokyo");
        map.put("MIT", "Pacific/Apia");
        map.put("NET", "Asia/Yerevan");
        map.put("NST", "Pacific/Auckland");
        map.put("PLT", "Asia/Karachi");
        map.put("PNT", "America/Phoenix");
        map.put("PRT", "America/Puerto_Rico");
        map.put("PST", "America/Los_Angeles");
        map.put("SST", "Pacific/Guadalcanal");
        map.put("VST", "Asia/Ho_Chi_Minh");
        map.put("EST", "-05:00");
        map.put("MST", "-07:00");
        map.put("HST", "-10:00");
        SHORT_IDS = Collections.unmodifiableMap(map);
    }
```

> If these values are hard coded in the kernel 
Yes. kernel hard coded, kernel handle EST/MST/HST with other fixed offsets together.
Fixed offsets, e.g.:
- UTC+08:00
- GMT-08:00
- -08:00
- +08:00
- EST/MST/HST
EST/MST/HST do not have time zone index. kernel handle them directly.

> can we verify that they are the only ones that we find here? 
Yes, the above java code listed the whole items. Only EST/MST/HST are fixed offsets.

> and that the offsets are what we expect them to be?
Yes.
EST: -05:00; HST: -07:00; MST: -10:00

```
>>> spark.conf.set("spark.sql.session.timeZone", "UTC")
>>> spark.sql("select cast('1970-01-01T00:00:00 EST' as timestamp)").show()
+------------------------------------------+
|CAST(1970-01-01T00:00:00 EST AS TIMESTAMP)|
+------------------------------------------+
|                       1970-01-01 05:00:00|
+------------------------------------------+
```

Reply:
> The only issue I have left that I think is a blocker is verifying that we have the short IDs properly covered when we build the data base.

Yes, they are 3 types:
- DST short IDs:  like unsupported regional based TZ. Should distinguish unsupported and invalid.
   I suggest move Plugin PR to 24.04: https://github.com/NVIDIA/spark-rapids/pull/10237
   Because it's a imcomplete PR. Strings can can contain unsupported TZs.
- fixed offset: EST/MST/HST: Kernel directly handle.
- supported regional based TZ.